service/internal/telemetrylogs/logger.go

```diff
@@ -45,5 +49,42 @@ func NewLogger(cfg config.ServiceTelemetryLogs, options []zap.Option) (*zap.Logg
 	if err != nil {
 		return nil, err
 	}
+
 	return logger, nil
 }
+
+// SettableGRPCLoggerV2 sets grpc framework's logger with internal logger.
+type SettableGRPCLoggerV2 interface {
+	SetGRPCLogger()
+}
+
+type colGRPCLogger struct {
+	setOnce  sync.Once
+	loggerV2 grpclog.LoggerV2
+}
+
+// NewColGRPCLogger constructs a grpclog.LoggerV2 instance cloned from baseLogger with exact configuration.
+// The minimum level of gRPC logs is set to WARN should the loglevel of the collector is set to INFO to avoid
+// copious logging from grpc framework.
+func NewColGRPCLogger(baseLogger *zap.Logger, loglevel zapcore.Level) SettableGRPCLoggerV2 {
+	logger := baseLogger.WithOptions(zap.WrapCore(func(core zapcore.Core) zapcore.Core {
+		var c zapcore.Core
+		if loglevel == zap.InfoLevel {
+			c, _ = zapcore.NewIncreaseLevelCore(core, zap.WarnLevel)
```

Comment:
I think ignoring this error deserves a comment; I would personally feel more comfortable if we return the error instead (since the core may change in the future and this may start returning an error then)

Reply:
Done! added a comment. Please let me know.

Reply:
👍 since the error conditions are documented on the function I think we can leave it like this, thanks